core/src/main/java/fr/sncf/osrd/conflicts/Conflicts.kt

```diff
@@ -204,6 +241,119 @@ class IncrementalConflictDetectorImpl(trainRequirements: List<TrainRequirements>
         }
         return res
     }
+
+    override fun minDelayWithoutConflicts(
+        spacingRequirements: List<SpacingRequirement>,
+        routingRequirements: List<RoutingRequirement>,
+        globalMinDelay: Double
+    ): Double {
+        var minDelay = Double.POSITIVE_INFINITY
+        var hasConflict = false
+        for (spacingRequirement in spacingRequirements) {
+            if (spacingZoneRequirements[spacingRequirement.zone!!] != null) {
+                val conflictingRequirements =
+                    spacingZoneRequirements[spacingRequirement.zone!!]!!.filter {
+                        !(spacingRequirement.beginTime >= it.endTime ||
+                            spacingRequirement.endTime <= it.beginTime)
+                    }
+                if (conflictingRequirements.isNotEmpty()) {
+                    hasConflict = true
+                    val latestEndTime = conflictingRequirements.maxOf { it.endTime }
+                    minDelay = min(minDelay, latestEndTime - spacingRequirement.beginTime)
+                }
+            }
+        }
+        for (routingRequirement in routingRequirements) {
+            for (zoneReq in routingRequirement.zones) {
+                if (routingZoneRequirements[zoneReq.zone!!] != null) {
+                    val conflictingRequirements =
+                        routingZoneRequirements[zoneReq.zone!!]!!.filter {
+                            !(routingRequirement.beginTime >= it.endTime ||
+                                zoneReq.endTime <= it.beginTime)
+                        }
+                    if (conflictingRequirements.isNotEmpty()) {
+                        hasConflict = true
+                        val latestEndTime = conflictingRequirements.maxOf { it.endTime }
+                        minDelay = min(minDelay, latestEndTime - routingRequirement.beginTime)
+                    }
+                }
+            }
+        }
+        if (!hasConflict) return globalMinDelay
+        else if (!isFinite(minDelay)) return Double.POSITIVE_INFINITY
+        else {
+            val shiftedSpacingRequirements =
+                spacingRequirements.toMutableList().onEach {
+                    it.beginTime += minDelay
+                    it.endTime += minDelay
+                }
+            val shiftedRoutingRequirements =
+                routingRequirements.toMutableList().onEach { routingRequirement ->
+                    routingRequirement.beginTime += minDelay
+                    routingRequirement.zones.onEach { it.endTime += minDelay }
+                }
+            return minDelayWithoutConflicts(
+                shiftedSpacingRequirements,
+                shiftedRoutingRequirements,
+                globalMinDelay + minDelay
+            )
```

Comment:
A comment explaining why we need a recursive call and why the values are shifted would help with readability

Reply:
Will replace it with a while loop as this is possible due to a previous comment about using max rather than min.